### PR TITLE
Moved macOsxVersion to appium-support + bump to 2.0.4

### PR DESCRIPTION
**Pull request by scottdixon**
appium-doctor todo https://github.com/appium/appium-doctor/issues/3

@imurchie

### CODE REVIEW

File: lib/system.js [hunk 1]
```diff
@@ -27,4 +27,17 @@ async function arch () {
   }
 }
 
-export { isWindows, isMac, isLinux, isOSWin64, arch };
+async function macOsxVersion () {
+  let stdout;
+  try {
+    stdout = (await exec('sw_vers', ['-productVersion'])).stdout;
+  } catch (err) {
+    throw new Error("Unknown SW Version Command: " + err);
```
Comment (imurchie) [inline lib/system.js]: Can we make a more intuitive error message? Like "Unable to detect Mac OS X version: XXX" or something?

Comment (scottdixon) [inline lib/system.js]: Good call. Line 40 should now take care of this - is there anything helpful we can bring out of line 35's error?

Comment (imurchie) [inline lib/system.js]: I think it would still be good to have this error thrown, since there is an error object to give the details of.

Comment (scottdixon) [inline lib/system.js]: I agree - but in regards to your 'intuitive' comment?

Comment (imurchie) [inline lib/system.js]: I would just use

```js
throw new Error(`Could not detect Mac OS X Version: ${err}`);
```

**Pull request merged**